### PR TITLE
User registration

### PR DESCRIPTION
**Pull request by NikitaNazarov1**
### What it does:
Add user registration, notification for successful registration and errors notification for unsuccessful registration (if something is not valid)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -25,3 +25,5 @@
 
 # Ignore master key for decrypting credentials and more.
 /config/master.key
+
+.idea/
```
Comment (labs-ruby) [inline .gitignore]: Environment-specific configs should be in your global gitignore

File: Gemfile [hunk 2]
```diff
@@ -4,6 +4,7 @@ source 'https://rubygems.org'
 git_source(:github) { |repo| "https://github.com/#{repo}.git" }
 
 ruby '2.6.3'
+# authorization
```
Comment (labs-ruby) [inline Gemfile]: Useless comment

File: app/assets/stylesheets/custom.scss [hunk 3]
```diff
@@ -0,0 +1,46 @@
+@import "bootstrap-sprockets";
```
Comment (labs-ruby) [inline app/assets/stylesheets/custom.scss]: Please, rename stylesheet to something specific

I don't like the idea to have "custom" css file

File: app/views/static_pages/root.html.erb [hunk 4]
```diff
@@ -6,7 +6,7 @@
   <div class="collapse navbar-collapse" id="navbarCollapse">
     <ul class="navbar-nav mr-auto"></ul>
     <form class="form-inline mt-2 mt-md-0">
-      <button class="form-control btn btn-outline-primary my-2 my-sm-0 mr-2 ml-2" type="submit">Sign Up</button>
+      <%= link_to "Sign up", signup_path, class: "form-control btn btn-outline-secondary my-2 my-sm-0 mr-2 ml-2" %>
```
Comment (labs-ruby) [inline app/views/static_pages/root.html.erb]: Don't hardcode text, use I18n

File: db/migrate/20190531120815_create_users.rb [hunk 5]
```diff
@@ -0,0 +1,12 @@
+class CreateUsers < ActiveRecord::Migration[5.2]
+  def change
+    create_table :users do |t|
+      t.string :first_name
+      t.string :last_name
+      t.string :email
```
Comment (labs-ruby) [inline db/migrate/20190531120815_create_users.rb]: We must add unique constraint to email on DB level!

File: app/views/static_pages/root.html.erb (hunk 4 above)
Comment (NikitaNazarov1) [reply app/views/static_pages/root.html.erb]: on adding I18n we have another task

File: db/schema.rb [hunk 6]
```diff
@@ -10,9 +10,18 @@
...
-ActiveRecord::Schema.define(version: 0) do
+ActiveRecord::Schema.define(version: 2019_05_31_120815) do
 
   # These are extensions that must be enabled in order to support this database
   enable_extension "plpgsql"
 
+  create_table "users", force: :cascade do |t|
+    t.string "first_name"
+    t.string "last_name"
+    t.string "email"
```
Comment (alexshgov) [inline db/schema.rb]: Where's the unique constraint? Did you apply a migration?

File: app/views/static_pages/root.html.erb (hunk 4 above)
Comment (BeOff) [reply app/views/static_pages/root.html.erb]: `i18n` with dictionaries are added now, please add translations if needed

**Comment by BeOff** [comment]
Please, add replies before resolving conversations.

**Comment by BeOff** [comment]
Please, do not use links in place of buttons. I. e. implement form on the root page with `form_with` if possible.

File: db/schema.rb (hunk 6 above)
Comment (NikitaNazarov1) [reply db/schema.rb]: Done, add index to field email

File: db/migrate/20190531120815_create_users.rb (hunk 5 above)
Comment (NikitaNazarov1) [reply db/migrate/20190531120815_create_users.rb]: done

File: .gitignore (hunk 1 above)
Comment (BeOff) [reply .gitignore]: Done!

File: app/assets/stylesheets/custom.scss (hunk 3 above)
Comment (BeOff) [reply app/assets/stylesheets/custom.scss]: Renamed!

File: app/views/static_pages/root.html.erb (hunk 4 above)
Comment (BeOff) [reply app/views/static_pages/root.html.erb]: Will do!

File: Gemfile (hunk 2 above)
Comment (BeOff) [reply Gemfile]: Will do!

**Pull request merged**